sdk/quota/arm-quota/samples/v1-beta/javascript/SetSubscriptionAllocationRequest.js

```diff
@@ -32,18 +31,65 @@ async function subscriptionQuotaAllocationPutRequestForCompute() {
   };
   const credential = new DefaultAzureCredential();
   const client = new AzureQuotaExtensionAPI(credential, subscriptionId);
-  const result = await client.groupQuotaSubscriptionAllocationRequest.beginCreateOrUpdateAndWait(
+  const result = await client.groupQuotaSubscriptionAllocationRequest.beginCreateOrUpdate(
```

Comment:
Could you share with me more background with this pr? Does arm-quota LRO follow our guideline: https://github.com/Azure/azure-resource-manager-rpc/blob/master/v1.0/async-api-reference.md?

Reply:
Yes, we were granted an exception for our API design because it doesn't follow the PUT operation pattern. When the operation completes for two of our APIs, the GET on the same resource will fail since that resource does not exist. We specified that the final state should be determined via location, but I believe the SDK is not using that. We have a tracking issue for this:https://github.com/Azure/azure-sdk-for-js/issues/30793

While we transition our PUT APIs to POST, to temporarily unblock customers we have updated these samples so they can achieve the same functionality as the CreateOrUpdateAndWait without it breaking.

Reply:
Thanks for this detail! Some follow-up questions:
- This is happened in latest preview version (2023-06-01-preview) and this version is already released, so we want to add samples to guide customers with correct resource, right?
- We plan to change the operation from PUT to POST in future, so AzureAsyncOperation + Location in POST will not be an issue for us, right?
- Also it means we could remove any customization in samples and it will be un-necessary, right?